jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -702,28 +702,7 @@ private void sendData(Request request, Response response, Callback callback, Htt
 
     protected void writeHttpContent(Request request, Response response, Callback callback, HttpContent content)
     {
-        try
-        {
-            ByteBuffer buffer = content.getByteBuffer(); // this buffer is going to be consumed by response.write()
-            if (buffer != null)
-            {
-                response.write(true, buffer, callback);
-            }
-            else
-            {
-                IOResources.copy(
-                    content.getResource(),
-                    response, request.getComponents().getByteBufferPool(),
-                    request.getConnectionMetaData().getHttpConfiguration().getOutputBufferSize(),
-                    request.getConnectionMetaData().getHttpConfiguration().isUseOutputDirectByteBuffers(),
-                    callback);
-            }
-        }
-        catch (Throwable x)
-        {
-            content.release();
-            callback.failed(x);
-        }
+        content.writeTo(response, callback);
```

Comment:
maybe
```suggestion
        content.writeTo(response, 
            new ByteBufferPool.Sized(
                request.getComponents().getByteBufferPool(),
                request.getConnectionMetaData().getHttpConfiguration().getOutputBufferSize(),
                request.getConnectionMetaData().getHttpConfiguration().isUseOutputDirectByteBuffers()
            ),
            callback);
```

Better yet, add a `request.getComponents().getSizedByteBufferPool()` that could return a pool per connection rather than making the wrapper every time?